airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -44,6 +44,7 @@ class DatetimeBasedCursor(StreamSlicer):
         partition_field_start (Optional[str]): partition start time field
         partition_field_end (Optional[str]): stream slice end time field
         lookback_window (Optional[InterpolatedString]): how many days before start_datetime to read data for (ISO8601 duration)
+        cursor_format (str): datetime format of the cursor field in the data record if different from `datetime_format`.
```

Comment:
Why would we have a different format between the cursor and the datetime in the stream state?

Reply:
I am trying to do the same in Low-Code.
But In low-code I need do this conversion in 2 places:

1. DatetimeStreamSlicer
3. Transformer (https://github.com/airbytehq/airbyte/blob/a8f7109a43f0cdfc7ed0127c038d749b08723fd1/airbyte-integrations/connectors/source-twilio/source_twilio/manifest.yaml#L59)

Reply:
So it's only for backward compatibility because for Twilio the state has the information as `2023-01-19T23:30:52Z` and not as `Thu, 19 Jan 2023 23:30:52 +0000` anymore, is that a fair summary? Else, we would probably not have to do this, correct?

Reply:
No, it's not related to the backward compatibility of `source-twilio`.

The current fundamental problem for me is that `DatetimeBasedCursor` works internally with `datetime` cursors as strings. If that `datetime` string is naturally sortable all works ok, but as only the cursor value is not sortable like in `source-twilio` (i.e. `"Thu, 19 Jan 2023 23:30:52 +0000"`) we need to do something with it.

[here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py#L105) we compare just simple strings

In this PR with the help of `cursor_datetime_format` I am trying to change: "Thu, 19 Jan 2023 23:30:52 +0000" -> "2023-01-19T23:30:52Z" inside `DatetimeBasedCursor` object to make things work.

Reply:
We can think about it in such a way:
This PR add a possibility to re-format from `cursor datetime_format` -> `datetime_format` internally.

Reply:
But yes I agree that this PR increases complexity by adding even more format options for this component.
I think we need fully re-design `DatetimeBasedCursor` and work internally with DateTime objects (not string)

Reply:
+1 for modifying DatetimeBasedCursor to handle Datetime objects instead of string. I think this will be more manageable than maintaining another format